lib/src/vector_math/quaternion.dart

```diff
@@ -394,6 +394,18 @@ class Quaternion {
         w * ow - x * ox - y * oy - z * oz);
   }
 
+  /// Check if two quaternions are the same.
+  @override
+  bool operator ==(Object other) =>
+      (other is Quaternion) &&
+      (_qStorage[0] == other._qStorage[0]) &&
```

Comment:
Compare in decreasing order of index - this helps array bounds check elimination (see `operator *` above)